### PR TITLE
Wanted

### PR DESCRIPTION
**Pull request by munchypizza**
-Removed passive hunger.
-Gave sparing a shorter cooldown and cost less hunger.
-Removed ghosts being slow.
-You lose half of your slime when !renouncing.
-You can !enlist from anywhere again.
-Now, instead of losing all your items upon death, you only loose some of them. Specifically, on death you..
--Loose ALL of your slime, levels, mutations, bounty, and unequipped weapons.
--Loose SOME of your food (half), unadorned cosmetics (half), current weaposkill (increased the max you're able to revive with from 3 to 6)
--Loose NONE of your adorned cosmetics and your equiped weapon, they are both just unequiped and you'll have to reequip them.

So, if you're a gangster and you kill another gangster, the above happens. But, if you're a gangster and you kill a Juvie, then you become WANTED, and a few things happen.
-While WANTED...
--Your information (slimes, level, weapon, weaponskill, location, hunger, etc.) are posting in your rival gang's base.
-You cannot enter or goto sub-zones.
-Your movement speed is reduced to half of the standard walking speed and you cannot teleport.
-You drop 100% of your SlimeCoin on death.
-Your teammates cannot kill you, nor can you suicide.
-You drop all your items on death.
-You loose all your weaponskills.
-All of the above lasts for one hour. All of your current bounty is converted into seconds and added to this timer.

### CODE REVIEW

File: ew.py [hunk 1]
```diff
@@ -140,43 +142,64 @@ def change_slimes(self, n = 0, source = None):
...
+
+				ewitem.item_dropsome(id_server = self.id_server, id_user = self.id_user, item_type_filter = ewcfg.it_cosmetic, fraction = cosmetic_fraction) # Drop a random fraction of your unadorned cosmetics on the ground.
+				ewitem.item_dedorn_cosmetics(id_server = self.id_server, id_user = self.id_user) # Unadorn all of your adorned hats.
+
+				ewitem.item_dropsome(id_server = self.id_server, id_user = self.id_user, item_type_filter = ewcfg.it_weapon, fraction = 1) # Drop random fraction of your unequipped weapons on the ground.
+				ewutils.weaponskills_clear(id_server = self.id_server, id_user = self.id_user, weaponskill = ewcfg.weaponskill_max_onrevive)
+
+			self.life_state = ewcfg.life_state_corpse
+			self.poi_death = self.poi
+			self.poi = ewcfg.poi_id_thesewers
```
Comment (magicksRampage) [inline ew.py:193]: ghosts should also be sent back to the sewers

File: ewcfg.py [hunk 2]
```diff
@@ -357,7 +373,7 @@
 }
 
 # Commands
-cmd_prefix = '!'
+cmd_prefix = '*'
```
Comment (magicksRampage) [inline ewcfg.py]: change this back

File: ewcosmeticitem.py [hunk 3]
```diff
@@ -100,17 +100,17 @@ def __init__(
 				item.item_props['adorned'] = 'false'
 				response = "You successfully dedorn your " + item.item_props['cosmetic_name'] + "."
 			else:
-				if adorned_items >= ewutils.max_adorn_bylevel(user_data.slimelevel):
-					response = "You can't adorn anymore cosmetics."
-				else:
-					item.item_props['adorned'] = 'true'
+				#if adorned_items >= ewutils.max_adorn_bylevel(user_data.slimelevel):
+				#	response = "You can't adorn anymore cosmetics."
+				#else:
```
Comment (magicksRampage) [inline ewcosmeticitem.py]: no more adorn limit?

File: ewitem.py [hunk 4]
```diff
@@ -357,6 +357,65 @@ def item_dropall(
...
+	if item_type_filter == ewcfg.it_cosmetic:
+		for item in drop_candidates:
+			cosmetic_id = item.get('id_item')
+			cosmetic_item = EwItem(id_item = cosmetic_id)
+			print("Adorned = {}".format(cosmetic_item.item_props['adorned']))
+			if cosmetic_item.item_props['adorned'] == "false":
+				filtered_items.append(item)
+				print("Filtered Items = {}".format(filtered_items))
+
+			if hasattr(cosmetic_item.item_props, 'slimeoid'):
```
Comment (magicksRampage) [inline ewitem.py]: hasattr checks attributes, not dictionary entries. use item_props.get('slimeoid') as a safe way to access that entry

File: ewitem.py [hunk 5]
```diff
@@ -357,6 +357,65 @@ def item_dropall(
...
 
+"""
+	Drop some of a player's non-soulbound items into their district.
+"""
+def item_dropsome(id_server = None, id_user = None, item_type_filter = None, fraction = None):
+	#try:
+	user_data = EwUser(id_server = id_server, id_user = id_user)
+	items = inventory(id_user = id_user, id_server = id_server, item_type_filter = item_type_filter)
+
+	print("Items = {}".format(items))
```
Comment (magicksRampage) [inline ewitem.py]: remove debug prints from this method

File: ewitem.py (hunk 4 above)
Comment (magicksRampage) [reply ewitem.py]: also you'll want to use an elif here, otherwise you might add the same item to the list twice

File: ewjuviecmd.py [hunk 6]
```diff
@@ -57,47 +57,63 @@ def __init__(self, grid):
...
+				ewcfg.cmd_enlist, user_data.faction, ewcfg.cmd_pardon)
+			return await ewutils.send_message(cmd.client, cmd.message.channel, ewutils.formatMessage(cmd.message.author, response))
+
+		response = "Enlisting in the {}.".format(ewcfg.faction_killers)
+		user_data.life_state = ewcfg.life_state_enlisted
+		user_data.faction = ewcfg.faction_killers
+		user_data.time_lastenlist = time_now + ewcfg.cd_enlist
+		user_data.persist()
+
+		if desired_faction == ewcfg.faction_rowdys:
```
Comment (magicksRampage) [inline ewjuviecmd.py]: formatting error. this will never be true

File: ewjuviecmd.py [hunk 7]
```diff
@@ -116,17 +132,19 @@ def __init__(self, grid):
...
 		response = "To turn in your badge, you must return to your soon-to-be former gang base."
 
 	else:
+		renounce_fee = user_data.slimes / 2
 		faction = user_data.faction
 		user_data.life_state = ewcfg.life_state_juvenile
 		user_data.weapon = -1
 		user_data.persist()
-		response = "You are no longer enlisted in the {}, but you are not free of association with them.".format(faction)
+		user_data.slimes = user_data.slimes - renounce_fee
```
Comment (magicksRampage) [inline ewjuviecmd.py]: use the user_data.change_slimes() method instead. also do this before you persist the object

File: ewmap.py [hunk 8]
```diff
@@ -28,6 +28,7 @@
 """
 def poi_is_pvp(poi_name = None):
 	poi = ewcfg.id_to_poi.get(poi_name)
+	poi = ewcfg.id_to_poi.get(poi_name)
```
Comment (magicksRampage) [inline ewmap.py]: duplicate line

File: ewmap.py [hunk 9]
```diff
@@ -800,6 +815,11 @@ def retrieve_locked_districts(id_server):
 	if inaccessible(user_data = user_data, poi = poi):
 		return await ewutils.send_message(cmd.client, cmd.message.channel, ewutils.formatMessage(cmd.message.author, "You're not allowed to go there (bitch)."))
 
+	# If you're WANTED, you can't enter sub-zones.
+	if user_data.time_expirpvp >= time_now:
+		if poi.is_pvp == False:
```
Comment (magicksRampage) [inline ewmap.py]: shouldn't we use is_subzone instead? is_pvp also applies to gangbases

File: ewwep.py [hunk 10]
```diff
@@ -671,6 +691,15 @@ def canAttack(cmd):
...
 			if was_shot:
 
+				if was_juvenile == True:
+					# Flag the shooter as WANTED no matter what happens next.
+					user_data.time_expirpvp = time_now + ewcfg.time_pvp_kill + int(user_data.bounty)
+
+					user_data.persist()
+
+					# Add the PvP flag role.
+					await ewrolemgr.updateRoles(client = cmd.client, member = cmd.message.author)
```
Comment (magicksRampage) [inline ewwep.py:701]: we'll need to reload a bunch of database objects after this await call

File: ewwep.py [hunk 11]
```diff
@@ -922,7 +950,81 @@ def canAttack(cmd):
...
+						weapon_item = EwItem(id_item = user_data.weapon)
+						weapon = ewcfg.weapon_map.get(weapon_item.item_props.get("weapon_type"))
+						descriptor = weapon.str_killdescriptor
+
+					else:
+						descriptor = "killed"
+
+					response += "wanted criminal {} has {} {} in {}! ".format(cmd.message.author.display_name, descriptor, member.display_name, ewmap.poi_id_to_display_name(user_data.poi))
+
+					response += "{} are a level {} slimeboi, and they are currently in possession of {:,} slime. ".format(cmd.message.author.display_name, user_data.slimelevel, user_data.slimes)
```
Comment (magicksRampage) [inline ewwep.py]: are -> is

Comment (magicksRampage) [reply ewwep.py]: the first one I mean

File: ewwep.py [hunk 12]
```diff
@@ -1429,8 +1538,8 @@ def canAttack(cmd):
 	elif weapon is None:
 		response = "How do you plan to get married to your weapon if you aren’t holding any weapon? Goddamn, think these things through, I have to spell out everything for you."
 		await ewutils.send_message(cmd.client, cmd.message.channel, ewutils.formatMessage(cmd.message.author, response))
-	#Makes sure you have a displayed rank 6 or higher weapon.
-	elif user_data.weaponskill < 10:
+	#Makes sure you have a displayed rank 4 or higher weapon.
+	elif user_data.weaponskill < 8:
```
Comment (magicksRampage) [inline ewwep.py:1544]: it still says rank 8 mastery in the flavor text

**Review by magicksRampage — CHANGES_REQUESTED** [review]

File: ewcosmeticitem.py [hunk 13]
```diff
@@ -100,10 +100,10 @@ def __init__(
...
-				#if adorned_items >= ewutils.max_adorn_bylevel(user_data.slimelevel):
-				#	response = "You can't adorn anymore cosmetics."
-				#else:
-				item.item_props['adorned'] = 'true'
+				if adorned_items >= ewutils.max_adorn_bylevel(user_data.slimelevel):
+					response = "You can't adorn anymore cosmetics."
+				else:
+					item.item_props['adorned'] = 'true'
 
 				if item.item_props.get('slimeoid') == 'true':
```
Comment (magicksRampage) [inline ewcosmeticitem.py]: this part will overwrite the adorn limit response

File: ewjuviecmd.py (hunk 6 above)
Comment (magicksRampage) [reply ewjuviecmd.py]: not fixed. this whole block is inside the killers block. desired_faction has already been confirmed to be killers, when this check is reached

**Review by magicksRampage — CHANGES_REQUESTED** [review]

File: ewcosmeticitem.py (hunk 3 above)
Comment (mudkipslaps) [reply ewcosmeticitem.py]: Is DTK getting a rework for this?

File: ewmap.py (hunk 9 above)
Comment (mudkipslaps) [reply ewmap.py]: I belive that is the intention?

**Pull request merged**